packages/grafana-runtime/src/services/dashboard-srv/DashboardSrvSingleton.ts

```diff
@@ -0,0 +1,98 @@
+import { PluginsAPIPanelModelWrapper } from './PanelWrapper';
+import {
+  CoreDashboardModel,
+  CoreDashboardSrv,
+  CorePanelModel,
+  PluginsAPIDashboardModel,
+  PluginsAPIDashboardSrv,
+  PluginsAPIPanelModel,
+} from './types';
+
+export class PluginsAPIDashboardModelWrapper implements PluginsAPIDashboardModel {
+  #dashboard: CoreDashboardModel | undefined;
```

Comment:
@dprokop this private field is not about sugar syntax is about the protection it offers in runtime of preventing access to the attribute.

the `private` keyword in typescript only prevents you from compiling code that tries to access an attribute tagged as private and that can be easily skipped by using a `//@ts-ignore`. In all the places I used hashed private fields I did it to prevent the plugin developer have access to the actual grafana-core object and use APIs that we don't want to expose.

Other techniques to achieve this is to have all the objects defined as variables in the module and use them as part of the scope of the classes but private fields do exactly this job and it makes it clear to know where an object is stored.